### PR TITLE
feat: added state(context api)

### PR DESCRIPTION
**Pull request by SakaRicky**

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -6,17 +6,20 @@ import { ThemeProvider } from '@material-ui/styles';
 import Layout from 'components/Layout';
```
Comment (lkuz) [inline pages/_app.tsx]: use named import 

`import { Layout } from 'components'`

File: state/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export * from 'state/reducer';
+export * from 'state/state';
+export * from 'state/stateTypes';
```
Comment (lkuz) [inline state/index.ts]: named exports - `export { reducer, state, stateTypes } from 'state'`

File: state/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import {  LoggedUser } from "types";
+import { State } from 'state/state';
```
Comment (lkuz) [inline state/reducer.ts]: State can be exported as `const State` and imported from 'state'

`import { State } from 'state' `

**Review by lkuz — DISMISSED** [review]

File: state/index.ts (hunk 2 above)
Comment (SakaRicky) [reply state/index.ts]: This create a circular import export problem.
Re-exporting type is done differently from others.

**Review by lkuz — DISMISSED** [review]

**Review by lkuz — APPROVED** [review]

**Pull request merged**